docs/intro.rst

```diff
@@ -124,66 +124,193 @@ In addition to samples and their data, collections may contain
    are groups of Data objects originating from the same biological
    sample—all Data objects in a sample are derived from a single NGS
    reads file. Collections are arbitrary groups of samples
    and Data objects that store analysis results.
 
-Samples, collections and studies have their own annotations, and
-the Data objects they contain have separate annotations. An example
-of the annotation field on a Study is ``organism`` and an example
-of the annotation field on a Sample is a ``barcode``. The annotation
-for sample (also collection and study) represents all Data objects
-attached to it.
-
 .. TODO: Image where clear distinction between Resolwe models /
          endpoints and resdk classes is presented.
 
+Annotations
+===========
+
+Annotations are presented as descriptors, where each descriptor is defined in
+descriptor schemas. Annotations for data objects, samples and collections are
+following different descriptor schemas. For example reads data object can be
+annotated with 'reads' descriptor schema, while sample can be annotated by
+'sample-geo' annotation schema. Each data object that is part of sample is
+also connected to sample annotation, so that the annotation for sample (also
+collection) represents all Data objects attached to it. Examples of descriptors
+and descriptor schemas are described in details in `Resolwe Bioinformatics
+documentation`_.
+
+.. _Resolwe Bioinformatics documentation: http://resolwe-bio.readthedocs.io
+
+Here we show how to annotate reads data object by defining descriptor:
+
+.. code-block:: python
+
+    annotation_reads = {
+        'experiment_type': 'RNA-seq',
+        'protocols': {
+            'growth_protocol': 'my growth protocol',
+            'treatment_protocol': 'my treatment protocol',
+            'library_prep': 'lybrary construction protocol',
+        },
+        'reads_info': {
+            'seq_date': '2016-10-13',
+            'instrument_type': 'Illumina',
+            'facility': 'my favorite facility',
+        }
+    }
+
+We can now annotate ``reads`` data object by adding descriptor and descriptor schema:
+
+.. code-block:: python
+
+    #define the chosen descriptor schema,
+    reads.descriptor_schema = 'reads'
+
+    #define the reads descriptor, with
+    reads.descriptor = annotation_reads
+
+    #save the annotation
+    reads.save()
+
+We can also define descriptor and descriptor schema directly when calling
+'res.run' function as described in The run method section.
+
+Managing presamples and samples
+===============================
+
+When you upload new data that belongs to the biological sample or run analysis
+on such data, process automatically create new (pre)sample for created object
+or add it to the an existing sample. For example if we upload fastq reads file
+and annotate presample to become sample, aligned bam and bai files will be also
+part of this same sample.
+
+When new sample is created, it is called ``presample``, which means that it is
```

Comment:
Unannotated samples (presamples) have a ``presample``property set to ``True``. To annotate a presample, fill its descriptor ...